From 3c36e2a3fec91f0eb1e58343d1d9dc3cf2ef2954 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Thu, 8 Sep 2016 22:08:07 +0700
Subject: [PATCH] Add GPIO button for RZG1M Starter Kit

Add GP7_0, GP7_1, GP7_2 into device tree for GPIO button of RZG1M Starter Kit
For DoorPhone on RZ/G1M Starter Kit, the virtual keyboard does not appear when the input email panel is clicked
Reason of this issue is RZ/G1M Starter Kit lacks off GPIO event
This patch add GPIO button into the device tree for making GPIO event to solve this issue

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Update-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 arch/arm/boot/dts/r8a7743-skrzg1m.dts | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743-skrzg1m.dts b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
index e0790f8..ec7362f 100644
--- a/arch/arm/boot/dts/r8a7743-skrzg1m.dts
+++ b/arch/arm/boot/dts/r8a7743-skrzg1m.dts
@@ -22,6 +22,7 @@
 /dts-v1/;
 #include "r8a7743.dtsi"
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>

 / {
	model = "Skrzg1m";
@@ -72,6 +73,32 @@
                };
	};

+	gpio-keys {
+		compatible = "gpio-keys";
+
+		key-a {
+			gpios = <&gpio7 0 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_A>;
+			label = "SW30";
+			gpio-key,wakeup;
+			debounce-interval = <20>;
+		};
+		key-b {
+			gpios = <&gpio7 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_B>;
+			label = "SW31";
+			gpio-key,wakeup;
+			debounce-interval = <20>;
+		};
+		key-c {
+			gpios = <&gpio7 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_C>;
+			label = "SW32";
+			gpio-key,wakeup;
+			debounce-interval = <20>;
+		};
+	};
+
	vcc_sdhi0: regulator@0 {
		compatible = "regulator-fixed";

--
1.9.1
